crates/macros/src/lib.rs

```diff
@@ -0,0 +1,62 @@
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{self, parse_macro_input, Data, DeriveInput, Error, Fields, Result};
+
+#[proc_macro_derive(BlockData)]
+pub fn block_data_derive(input: TokenStream) -> TokenStream {
+    let ast = parse_macro_input!(input as DeriveInput);
+
+    impl_block_data(&ast).unwrap_or_else(|err| err.to_compile_error().into())
+}
+
+fn impl_block_data(ast: &DeriveInput) -> Result<TokenStream> {
+    let name = &ast.ident;
+    let field;
+
+    if let Data::Struct(data) = &ast.data {
+        if let Fields::Named(named_fields) = &data.fields {
+            if named_fields.named.len() != 1 {
```

Comment:
I think we need support for multiple fields, since most of Spigot's `BlockData`s do provide multiple properties.

Reply:
Updated to support multiple fields